### PR TITLE
track-width slider 

### PR DESCRIPTION
**Pull request by TeunHuijben**
I implemented an extra UI slider for the user to change the width of the tracks (and highlights)

The chosen implementation does: 
- the width of tracks and track highlights is connected to the pointSize of cells
- the track highlights have a width equal to the cell size
- the tracks itself have a width `7x` thinner than the cell size
- if radius is provided, the cell size is taken as the average cell radius in the first frame is chosen (computed only once)
- the track-width slider starts at `100%` and scales the width of the tracks and the highlights
- if the cell-size-slider is moved, the tracks/highlight scale accordingly
- track-width and track-highlight-width sliders disappear when tracks/highlights are toggled off
- `pointSize` and `trackWidthFactor` are saved in `ViewerState`, so they are saved/loaded when state if copied into an URL

https://github.com/user-attachments/assets/aeebc12d-c6a3-4739-beeb-a9a0b0b22ddb

### CODE REVIEW

File: src/lib/PointCanvas.ts [hunk 1]
```diff
@@ -76,6 +76,10 @@ export class PointCanvas {
     maxTime: number = 5;
     pointBrightness = 1.0;
     pointSize = initialPointSize;
+    trackWidthFactor = 1; // changed by track-width slider
+    trackWidthRatio = 0.07; // DONT CHANGE: factor of 0.07 is needed to make tracks equally wide as the points
```
Comment (andy-sweet) [inline src/lib/PointCanvas.ts:80]: If this is never supposed to change, I would define outside of the class as 

```typescript
// Width ratio of 0.07 is needed to make tracks equally wide as the points
const trackWidthRatio = 0.07;
```

Comment (andy-sweet) [reply src/lib/PointCanvas.ts:80]: Similar comment for the next two variables.

File: src/lib/ViewerState.ts [hunk 2]
```diff
@@ -24,8 +25,11 @@ export class ViewerState {
     showTracks = true;
     showTrackHighlights = true;
     // Default position and target for the camera
-    cameraPosition = [0, 0, -4]; // was  [500, 500, -1250];
-    cameraTarget = [0, 0, 0]; // was  [500, 500, 500];
+    cameraPosition = [0, 0, -4];
+    cameraTarget = [0, 0, 0];
+    pointSize: number = initialPointSize;
+    trackWidth: number = 0.5;
```
Comment (andy-sweet) [inline src/lib/ViewerState.ts:31]: Is this ever used?

**Review by andy-sweet — COMMENTED** [review]
One problem I found when using this branch is that if I change the track width while tracks are loading, only the loaded tracks' width is changed and the ones loaded later get the old track width value.

I think that might be an issue for other parameters and it is easy enough to workaround by changing the value when all tracks have loaded, so this is non-blocking.

File: src/lib/ViewerState.ts (hunk 2 above)
Comment (TeunHuijben) [reply src/lib/ViewerState.ts:31]: They are used in `PointCanvas.ts/toState`, to ensure these settings are taken into the hash of the configuration state, when copied as URL.

Comment (andy-sweet) [reply src/lib/ViewerState.ts:31]: I see that `trackWidthFactor` is returned by `toState` and used for rendering, but don't see the same for `trackWidth`.

**Pull request merged**